homeassistant/components/recorder/__init__.py

```diff
@@ -1116,9 +1116,18 @@ def _commit_event_session(self):
                 if dbstate in self.event_session:
                     self.event_session.expunge(dbstate)
             self._pending_expunge = []
-        self._pending_state_attributes = {}
         self.event_session.commit()
 
+        # We just committed the state attributes to the database
+        # and we now know the attributes_ids.  We can save
+        # a many selects for matching attributes by loading them
```

Comment:
We can save many selects

Reply:
I fixed it in my typing cleanup branch. I pulled out just the `__init__.py` file into a smaller PR: https://github.com/home-assistant/core/pull/68551